### ISSUE TITLE
🎯 Roadmap

### ISSUE THREAD

**Issue by visortelle** [visortelle/dekaf#141]
Feel free to create a separate issue for each of them.

Items marked as 🎯 we should finish until the initial release.

### App

- [x] #144
- [x] #145
- [x] #149
- [ ] 🎯 Add more cases to the demoapp. It would allow us to manually test common cases.
  - [x] 🎯 Fix perf issues with high-freq STRING and BYTES topics data generation.
  - [x] 🎯 Add a demo for the AVRO schema.
  - [x] 🎯 Add a demo for the PROTOBUF_NATIVE schema.
  - [ ] 🎯 Add a demo for each subscription type.
  - [ ] 🎯 Add a demo for each producer mode.
  - [ ] 🎯 Add a demo for a lot of namespaces per tenant and topics per namespace.
    - 🎯 Use persistent and non-persistent topics, partitioned and non-partitioned topics.
- [x] #146
- [x] #148
- [ ]  #153 
- [x] https://github.com/tealtools/pulsar-ui/issues/163
- [x] #169
- [ ] 🎯 For topics with low message rates like 1 mps, the order of the messages is shuffled on UI. It looks like the issue appeared after switching to `acknowledgeAsync`. We should re-check it and decide if sync acknowledge performance is enough, or we should do some additional sorting actions when we send or process the bunch of messages to the UI.
- [x] 🎯 Handle the situation where topic has an exclusive producer. See https://github.com/tealtools/pulsar-ui/issues/22
- [x] #147
  - [x] 🎯 Update descriptions for topic policies.
- [ ] 🎯 Fix or hide broken consumer view tabs like `Cursors`, `Filters REPL`.
- [ ]  #171 
- [ ] 🎯 Add Data masking 
- [ ] 🎯 Create separate versions of the App
  - [ ] Standalone Edition - Run pulsar-standalone instance + Pulsocat in a single Docker container
  - [ ] Paid Edition

### Sales

- [ ] 🎯 Implement site. Some initial work is already done here. https://github.com/tealtools/pulsar-ui-website
- [ ] 🎯 Document all the major features. For now, markdown on GitHub is enough. Later we'll build a static site using Docusaurus or a similar tool.
- [ ] 🎯 Deploy demoapp to AWS.
- [ ] 🎯 Build and push the dev-box Docker image with limited features set to DockerHub.
- [ ] 🎯 Build and push the Docker image to private repo at the DockerHub.
- [ ] 🎯 Setup Stripe and check workflow with manual DockerHub tokens issuing.

### Soon after the release

- [ ] Use sidebar instead of modal in consumer view

### Later

- [ ] Namespace deletion dialog stuck in case of a large number of topics in the namespace.
  Reproducible on 8k topics.
- [ ] Namespaces table. Getting the topics count is slow in case of some namespaces have large numbers of topics.
  - Currently, we request the topics count for a bunch of namespaces visible on the screen. After the user scrolls down the table, the topics count will be requested again for the next visible bunch of namespaces.
  - Then we are waiting for all the responses.
  - Then we send the batch response to the user.
- [ ] Handle big messages.
  
  I think we should consider streaming results per/namespace. We can use the server-side streaming grpc-web feature.
